src/App.jsx

```diff
@@ -14,33 +15,84 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const GoodList = ({ goods }) => (
+  <ul>
+    {goods.map(good => (
+      <li data-cy="Good" key={good}>
+        {good}
+      </li>
+    ))}
+  </ul>
+);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+function getPreparedGoods(goods, sortField) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortField === SORT_FIELD_ALPHABET) {
+    return preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  if (sortField === SORT_FIELD_LENGTH) {
+    return preparedGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  let visibleGoods = getPreparedGoods(goodsFromServer, sortField);
+  const [reversed, setReversed] = useState(false);
+
+  if (reversed) {
+    visibleGoods = [...visibleGoods].reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortField === SORT_FIELD_ALPHABET ? '' : 'is-light'}`}
```

Comment:
It's better to use classnames library to set classes conditionally
https://www.npmjs.com/package/classnames